frontends/web/src/routes/bitboxbase/bitboxbase.tsx

```diff
@@ -74,8 +75,26 @@ export interface BaseUpdateInfo {
     severity: string;
 }
 
+const initSharedBaseState = (baseID: string) => {
+    const initialStatus: BaseStatus = '';
+    const initialState = {[baseID]: {
+        status: initialStatus,
+        paired: false,
+    }};
+    baseStore.setState({registeredBases: initialState});
+};
+
+export const updateSharedBaseState = <K extends keyof RegisteredBaseFields>(key: K, value: RegisteredBaseFields[K], baseID: string) => {
+    if (!(baseID in baseStore.state.registeredBases)) {
+        initSharedBaseState(baseID);
+    }
+    const newState = Object.assign({}, baseStore.state);
+    newState.registeredBases[baseID][key] = value;
+    baseStore.setState(newState);
```

Comment:
In theory when you use current/previous state in setState you should pass a function to setState. The reason is that setState may be asynchronously and current/previous state might have changed in the meantime, so this probably won't matter.

https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

Reply:
Indeed, except in this case it's not the setState from React that's being called, but our custom setState from here: https://github.com/digitalbitbox/bitbox-wallet-app/blob/ed94890b7354c119cbde9111c79ac1720a65da13/frontends/web/src/decorators/store.ts#L56
which doesn't take a callback like setState in React and instead it calls `forceUpdate()` on each component. :man_shrugging: 
I'm open to suggestions :)

Reply:
ah yeah I forgot, in that case please ignore my comment. sorry!!!!

forceUpdate :(